node/pkg/driver/device_connectivity/device_connectivity_iscsi.go

```diff
@@ -48,10 +48,26 @@ var (
 	TimeOutMultipathFlashCmd = 4 * 1000
 )
 
-func (r OsDeviceConnectivityIscsi) RescanDevices(lunId int, arrayIdentifier string) error {
-	klog.V(5).Infof("Rescan : Start rescan on specific lun, on lun : {%v}, with array iqn : {%v}", lunId, arrayIdentifier)
-	sessionHosts, err := r.Helper.GetIscsiSessionHostsForArrayIQN(arrayIdentifier)
-	if err != nil {
+func (r OsDeviceConnectivityIscsi) RescanDevices(lunId int, arrayIdentifiers []string) error {
+	klog.V(5).Infof("Rescan : Start rescan on specific lun, on lun : {%v}, with array iqn : {%v}", lunId, arrayIdentifiers)
+	var sessionHosts []int
+	var err error
+
+	if len(arrayIdentifiers) == 0 {
+		e := &ErrorNotFoundArrayIdentifiers{lunId}
+		klog.Errorf(e.Error())
+		return e
+	}
+
+	for _, iqn := range arrayIdentifiers{
+		h, e := r.Helper.GetIscsiSessionHostsForArrayIQN(iqn)
+		if e != nil {
+			klog.Errorf(e.Error())
+			err = e
```

Comment:
What kind of errors are you expecting here?
There is an option that you will get multiple errors, or some disconnected sessions but you will still go forward to the next identifier, and you will still append the session. The logic is not clear to me, can you explain what you mean to do?
What is 'h'? can you use a more self explanatory variable name?

Reply:
I think we can ignore the errors when the length of sessionHosts is not 0.
for different scsi hosts(/sys/class/scsi_host/hostX/scan), they are just the different paths of the same physical device, 
so if get iscsi session hosts for one array iqn failed, we just need to continues with the next one.

if it's 0, I only returned the last one error here, it's not very resonable, it better to add all the error messages together.

for the h, I will update it, Thanks